Update stable with develop

robbibt opened the pull request:
### Proposed changes
This update will add the new Export_imagery.ipynb notebook to the stable branch so that it will be available on the DEA Sandbox

robbibt committed:
Update template and erosion notebooks to remove scripts

robbibt committed:
Rerun with smaller dask cluster

robbibt committed:
Update shipping lane detection

robbibt committed:
Remove ARD comparison (#869)

MatthewJA committed:
Add notebook test action

Commit:
Added Tests folder

robbibt committed:
Notebook for exporting high quality satellite images from an interactive map (#870)

* Rerun notebook

* Update tags

* Add option to export machine readable name

robbibt committed:
Add new functionality to Image export notebook and functions (#873)

* Make area selection independent of image export; allow jpg

* Re-run export notebook, add to readme

MatthewJA commented:
Maybe we could get the tests on nbtests passing first, then merge that to develop, then merge that to stable? Just wary that the current version of develop has a failing test suite.

BexDunn committed:
Update warnings (#878)

* Update warnings

Remove warning where Tasselled cap indices for landsat are not normalised. This addresses issue #877

* Remove remaining TCI indices from warning list

Co-authored-by: Robbi Bishop-Taylor <Robbi.BishopTaylor@ga.gov.au>

robbibt committed:
Fix image export for Sentinel-2 NRT (#876)

robbibt committed:
Add first set of basic Testbook tests to DEA Notebooks (#872)

* Update and rename test-notebooks.yml to test_notebooks.yml

* Update test_notebooks.yml

* Update test_notebooks.yml

* Update test_notebooks.yml

* Update test_notebooks.yml

* Update test_notebooks.yml

* Added test for beginners guide

* create testing notebook for no 7

* Updated tests

* Create docker-compose.yml

* Add env vars

* one test for 07

* Update test_notebooks.yml

* create 02_DEA_test.py as a copy of 01

* Update test_notebooks.yml

* runs tests on 02

* random typo broke 02; fixed

* copied 02 to 03 test

* point to the correct nb

* added postgres and index containers

* created loading data test file

* Add new frequently used code tests

* Fix relative file path errors in contour extraction

* Minor updates to notebooks to improve tests

* Add additional tests

* Fix relative paths, standardise test format using Black

* Update test_notebooks.yml

* Index geomedians

* Update setup_test_datacube.sh

* Update test_notebooks.yml

* Added sandbox image

* Added volume mount

* Docker compose

* Added metadata and product catalogues to index

* Update test_notebooks.yml

* Update test_notebooks.yml

* Installed testbook

* Install local dea-tools

* in-tree-build for dea-tools

* Copy Tools to writable directory before build

* Explicit home dir for Tools

* Update test_notebooks.yml

* Added scene getter script

* Added dask "support"

* Patched arbitary functions

* Sorted and removed duplicates

* Add scenes for 06_Basic_analysis_test

* Temporarily comment out failing tests and indexing paths

* Revert commenting out

* Ignore frequently used code and 06_Basic_analysis_test.py

* Also remove intro to numpy test due to relative link issue

* Use correct file name

Co-authored-by: Matthew Alger <matthew.alger@gmail.com>
Co-authored-by: Matthew Alger <matthew.alger@ga.gov.au>
Co-authored-by: BexDunn <bex.dunn@ga.gov.au>

BexDunn reviewed (approved):
wow, this one is a big one

robbibt committed:
Add testing badge (#880)

* Add testing badge

* Remove accidently committed text
